purchase_stock_price_variance/tests/test_purchase_stock_price_variance.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestPurchaseStockPriceVariance(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company = cls.env.user.company_id
+        cls.product_category = cls.env["product.category"].create(
+            {"name": "Test Category"}
+        )
+        cls.product = cls.env["product.template"].create(
+            {
+                "name": "Test Product",
+                "standard_price": 100,
+                "categ_id": cls.product_category.id,
+            }
+        )
+        cls.supplier = cls.env["res.partner"].create({"name": "Test Supplier"})
+        cls.group_manage_price_variance_check = cls.env.ref(
+            "purchase_stock_price_variance.group_manage_price_variance_check"
+        )
+        cls.company.enable_price_variance_error = True
+
+    def create_purchase_order(self, price_unit=100):
+        return self.env["purchase.order"].create(
+            {
+                "partner_id": self.supplier.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product.product_variant_id.id,
+                            "product_qty": 1,
+                            "price_unit": price_unit,
+                        },
+                    )
+                ],
+            }
+        )
+
+    def validate_picking(self, picking, expect_error=False):
+        picking.move_ids.write({"quantity_done": 1})
+        if expect_error:
+            with self.assertRaises(UserError):
+                picking.button_validate()
+        else:
+            picking.button_validate()
+            self.assertEqual(picking.state, "done")
+
+    def check_chatter_message(self, picking, should_contain):
+        messages = picking.message_ids.mapped("body")
+        found = any(
+            "Price variance exceeding a threshold detected" in message
+            for message in messages
+        )
+        self.assertEqual(found, should_contain)
+
+    def test_01_normal_workflow_no_error(self):
+        po = self.create_purchase_order()
+        po.button_confirm()
+        picking = po.picking_ids
```

Comment:
```suggestion
```

Reply:
I prefer using new variable and pass as a parameter.

Reply:
Sorry, I commented on the wrong line. I meant `self.assertTrue(picking)` below.